lib/spack/spack/test/spec_yaml.py

```diff
@@ -421,12 +424,14 @@ def test_load_json_specfiles(specfile, expected_hash, reader_cls):
     assert len(openmpi_edges) == 1
 
     # Check that virtuals have been reconstructed
-    assert "mpi" in openmpi_edges[0].virtuals
-
-    # The virtuals attribute must be a tuple, when read from a
-    # JSON or YAML file, not a list
-    for edge in s2.traverse_edges():
-        assert isinstance(edge.virtuals, tuple), edge
+    match = re.search(specfile_version_regex, specfile)
+    if match.group(1) >= "v020":
```

Comment:
Can we discriminate based on the reader instead, e.g. `SpecfileV4` etc., or if it is difficult, by adding another parameter to the test?

Reply:
Sure. I assume using the class' `SPEC_VERSION` is better anyway.

Reply:
Done.